include/seqan3/core/type_traits/iterator.hpp

```diff
@@ -148,89 +148,87 @@ struct size_type<it_t>
 #endif // SEQAN3_DEPRECATED_310
 
 // see specialisation for ranges in core/type_traits/range.hpp
+//!\}
 
-// ----------------------------------------------------------------------------
-// iterator_tag
-// ----------------------------------------------------------------------------
+} // namespace seqan3
 
+namespace seqan3::detail
+{
+#if SEQAN3_WORKAROUND_GCC_96070
+//!\cond
+template <typename it_t>
+struct iterator_category_tag
+{
+    using type = void;
+};
+
+template <typename it_t>
+    requires requires { typename std::iterator_traits<it_t>::iterator_category; }
+struct iterator_category_tag<it_t>
+{
+    using type = typename std::iterator_traits<it_t>::iterator_category;
+};
+//!\endcond
 /*!\brief Exposes the
  * [iterator_category](https://en.cppreference.com/w/cpp/iterator/iterator_tags) from the modelled concept.
  * \implements seqan3::transformation_trait
  * \tparam it_t The type to operate on.
- * \see seqan3::iterator_tag_t
  *
  * \attention
  * If [std::iterator_traits<it_t>::iterator_category](https://en.cppreference.com/w/cpp/iterator/iterator_traits)
- * is defined for a type `it_t`, this transformation trait acts as an alias for it.
- * If it is not defined and no concepts are modelled, iterator_tag<it_t>::type is not defined.
+ * is defined for a type `it_t`, this transformation trait acts as an alias for it, if not it is `void`.
  */
 template <typename it_t>
-struct iterator_tag
-{
-#if SEQAN3_DOXYGEN_ONLY(1)0
-    //!\brief The [iterator_category](https://en.cppreference.com/w/cpp/iterator/iterator_tags).
-    using type = iterator_category;
-#endif
-};
+using iterator_category_tag_t = typename iterator_category_tag<it_t>::type;
```

Comment:
sill this is documented as transformation trait but it is not since it does not have a member type called `type`. This is the shortcut for the actual transformation trait. Either remove the `\implements seqan3::transformation_trait` or move the documentation to the actual transformation trait and document this as the shortcut for this.